### PR TITLE
Use infrastructure Docker image.

### PR DESCRIPTION
**Pull request by valkov-ivaylo**
PT-161478102

Requires aeternity/infrastructure#147

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -38,15 +38,6 @@ references:
         sudo usermod -aG docker ubuntu
         sudo chown -R ubuntu:ubuntu ~/epoch
 
-  install_docker_client: &install_docker_client
-    run:
-      name: Install Docker CLI tools
-      command: |
-        curl -L -o /tmp/docker-${DOCKER_CLIENT_VERSION:?}.tgz \
```
Comment (lucafavatella) [inline .circleci/config.yml]: Is `DOCKER_CLIENT_VERSION: "17.09.0-ce"` in this file now unused and to be removed?

File: .circleci/config.yml [hunk 2]
```diff
@@ -504,18 +495,17 @@ jobs:
       - *fail_notification
 
   docker_push_tag:
-    <<: *container_config
+    <<: *infrastructure_config
```
Comment (lucafavatella) [inline .circleci/config.yml]: Requires new Docker image published. Ok.

File: .circleci/config.yml (hunk 1 above)
Comment (valkov-ivaylo) [reply .circleci/config.yml]: It might be removed. I did not find any other references it in in the .circleci/config.yml. I was not sure. @dincho ?

Comment (dincho) [reply .circleci/config.yml]: I don't see other references either, let's remove it

Comment (valkov-ivaylo) [reply .circleci/config.yml]: Fixed in 45e3b44

**Pull request merged**